src/app/modules/myinfo/myinfo.util.ts

```diff
@@ -0,0 +1,315 @@
+      }
+  }
+}
+
+export const mapRedirectURLError: MapRouteError = (
+  error,
+  coreErrorMessage = 'Something went wrong. Please refresh and try again.',
+) => {
+  switch (error.constructor) {
+    case FormNotFoundError:
+      return {
+        statusCode: StatusCodes.NOT_FOUND,
+        errorMessage:
+          'Could not find the form requested. Please refresh and try again.',
+      }
+    case MyInfoAuthTypeError:
+      return {
+        statusCode: StatusCodes.BAD_REQUEST,
+        errorMessage:
+          'This form does not have MyInfo enabled. Please refresh and try again.',
+      }
+    case DatabaseError:
+    case MissingFeatureError:
+      return {
+        statusCode: StatusCodes.INTERNAL_SERVER_ERROR,
+        errorMessage: coreErrorMessage,
+      }
+    default:
+      logger.error({
+        message: 'Unknown route error observed',
+        meta: {
+          action: 'mapRedirectURLError',
+        },
+        error,
+      })
+      return {
+        statusCode: StatusCodes.INTERNAL_SERVER_ERROR,
+        errorMessage: coreErrorMessage,
+      }
+  }
+}
+
+export const mapEServiceIdCheckError: MapRouteError = (
+  error,
+  coreErrorMessage = 'Something went wrong. Please refresh and try again.',
+) => {
+  switch (error.constructor) {
+    case FormNotFoundError:
+      return {
+        statusCode: StatusCodes.NOT_FOUND,
+        errorMessage:
+          'Could not find the form requested. Please refresh and try again.',
+      }
+    case MyInfoAuthTypeError:
+      return {
+        statusCode: StatusCodes.BAD_REQUEST,
+        errorMessage:
+          'This form does not have MyInfo enabled. Please refresh and try again.',
+      }
+    case MyInfoNoESrvcIdError:
+      return {
+        statusCode: StatusCodes.FORBIDDEN,
+        errorMessage: 'This form does not have a valid e-service ID.',
+      }
+    case FetchLoginPageError:
+    case LoginPageValidationError:
+      return {
+        statusCode: StatusCodes.SERVICE_UNAVAILABLE,
+        errorMessage: 'Failed to contact SingPass. Please try again.',
+      }
+    case DatabaseError:
+    case MissingFeatureError:
+    case CreateRedirectUrlError:
+      return {
+        statusCode: StatusCodes.INTERNAL_SERVER_ERROR,
+        errorMessage: coreErrorMessage,
+      }
+    default:
+      logger.error({
+        message: 'Unknown route error observed',
+        meta: {
+          action: 'mapRedirectURLError',
+        },
+        error,
+      })
+      return {
+        statusCode: StatusCodes.INTERNAL_SERVER_ERROR,
+        errorMessage: coreErrorMessage,
+      }
+  }
+}
+
+export const getMyInfoFieldOptions = (
+  myInfoAttr: IMyInfo['attr'],
+): string[] => {
+  const [myInfoField] = myInfoTypes.filter((type) => type.name === myInfoAttr)
+  return myInfoField?.fieldOptions || []
+}
+
+export const createConsentPagePurpose = (formTitle: string): string =>
+  `The form "${formTitle}" is requesting to pre-fill your MyInfo data.`
```

Comment:
I would personally prefer a static message such as "form.gov.sg is requesting to pre-fill your MyInfo data" to reinforce the legitimacy of the request and enforce consistent UX instead of relying on a variable-length form title - shall we have a quick discussion at team sync-up around this?

Reply:
changed to a static message, already agreed with @syan-syan